### PR TITLE
feat: Added db.defrag.enabled and db.wal.checkpoint.enabled fields.

### PR DESCRIPTION
**Pull request by salvatore-coppola**
Added db.defrag.enabled and db.wal.checkpoint.enabled fields and related options.

### CODE REVIEW

File: kura/org.eclipse.kura.db.sqlite.provider/OSGI-INF/metatype/org.eclipse.kura.db.SQLiteDbService.xml [hunk 1]
```diff
@@ -65,22 +65,38 @@
...
+            description="Enables or disables checkpoints in WAL journal mode."/>
+
          <AD id="db.wal.checkpoint.interval.seconds"
             name="WAL Checkpoint Interval (Seconds)"
             type="Long"
             cardinality="0" 
-            required="true"
+            required="false"
             default="600"
-            min="0"
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.db.sqlite.provider/OSGI-INF/metatype/org.eclipse.kura.db.SQLiteDbService.xml:83]: Maybe we can set one minute as minimum value also here

**Comment by nicolatimeus** [comment]
Do we want to add integration tests to verify that defrag and checkpoint can be disabled?

**Comment by salvatore-coppola** [comment]
> Do we want to add integration tests to verify that defrag and checkpoint can be disabled?

uhm, set up the integration test from scratch take some times (now there are not). Are we sure we are not fine with QA tests ?

**Comment by MMaiero** [comment]
@salvatore-coppola Are you working on this? Or are we gonna make another PR to improve the tests suggested by @nicolatimeus?

**Comment by salvatore-coppola** [comment]
> @salvatore-coppola Are you working on this? Or are we gonna make another PR to improve the tests suggested by @nicolatimeus?

i'm writing the tests just right now.

File: kura/org.eclipse.kura.db.sqlite.provider/OSGI-INF/metatype/org.eclipse.kura.db.SQLiteDbService.xml [hunk 2]
```diff
@@ -65,22 +65,38 @@
...
+
          <AD id="db.defrag.interval.seconds"
             name="Defrag Interval (seconds)"
             type="Long"
             cardinality="0" 
-            required="true"
+            required="false"
             default="900"
             min="60"
             description="SqliteDbService instances support running periodic defragmentation using the VACUUM command (https://www.sqlite.org/lang_vacuum.html). This parameter specifies the interval in seconds beetween two consecutive defragmentations, set to zero to disable. This parameter is only relevant for persisted databases."/>
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.db.sqlite.provider/OSGI-INF/metatype/org.eclipse.kura.db.SQLiteDbService.xml]: This no longer applies.

```suggestion
            description="SqliteDbService instances support running periodic defragmentation using the VACUUM command (https://www.sqlite.org/lang_vacuum.html). This parameter specifies the interval in seconds beetween two consecutive defragmentations. This parameter is only relevant for persisted databases."/>
```

File: kura/org.eclipse.kura.db.sqlite.provider/OSGI-INF/metatype/org.eclipse.kura.db.SQLiteDbService.xml [hunk 3]
```diff
@@ -65,22 +65,38 @@
...
          <AD id="db.wal.checkpoint.interval.seconds"
             name="WAL Checkpoint Interval (Seconds)"
             type="Long"
             cardinality="0" 
-            required="true"
+            required="false"
             default="600"
-            min="0"
+            min="60"
             description="SqliteDbService instances support running periodic periodic WAL checkpoints (https://www.sqlite.org/pragma.html#pragma_wal_checkpoint). Checkpoints will be performed in TRUNCATE mode. This parameter specifies the interval in seconds beetween two consecutive checkpoints, set to zero to disable. This parameter is only relevant for persisted databases in WAL Journal Mode."/>
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.db.sqlite.provider/OSGI-INF/metatype/org.eclipse.kura.db.SQLiteDbService.xml]: Same here
```suggestion
            description="SqliteDbService instances support running periodic periodic WAL checkpoints (https://www.sqlite.org/pragma.html#pragma_wal_checkpoint). Checkpoints will be performed in TRUNCATE mode. This parameter specifies the interval in seconds beetween two consecutive checkpoints. This parameter is only relevant for persisted databases in WAL Journal Mode."/>
```

File: kura/org.eclipse.kura.db.sqlite.provider/OSGI-INF/metatype/org.eclipse.kura.db.SQLiteDbService.xml (hunk 2 above)
Comment (nicolatimeus) [reply kura/org.eclipse.kura.db.sqlite.provider/OSGI-INF/metatype/org.eclipse.kura.db.SQLiteDbService.xml]: Maybe we can expand the description of the enable options for both defrag and checkpoint to specify how they are related with the interval parameters.

**Review by nicolatimeus — APPROVED** [review]

**Pull request merged**